### PR TITLE
Transition from Microsoft.Azure.KeyVault to Azure.Security.KeyVault

### PR DESCRIPTION
**Pull request by erdembayar**
Related to: https://github.com/NuGet/Engineering/issues/4704

Transition from Microsoft.Azure.KeyVault to Azure.Security.KeyVault package dependency. Former one is deprecated.
I manually tested this change in [Gallery dev environment](https://devdiv.visualstudio.com/DevDiv/_releaseProgress?_a=release-pipeline-progress&releaseId=1478870)

### CODE REVIEW

File: src/DatabaseMigrationTools/DatabaseMigrationTools.csproj [hunk 1]
```diff
@@ -64,8 +64,9 @@
       <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
+    <!-- TODO: Update with stable version later -->
```
Comment (joelverhagen) [inline src/DatabaseMigrationTools/DatabaseMigrationTools.csproj:67]: Sorry, I mean a GitHub comment not an inline comment. No worry though, this works just is more work.

**Review by joelverhagen — APPROVED** [review]
:shipit: - I'll need to re-approve after changes though.

File: src/DatabaseMigrationTools/DatabaseMigrationTools.csproj [hunk 2]
```diff
@@ -65,7 +65,7 @@
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
     <PackageReference Include="NuGet.Services.Validation">
-      <Version>2.109.0</Version>
+      <Version>2.110.0-agr-kv-lib-upgrade3-8361872</Version>
```
Comment (erdembayar) [inline src/DatabaseMigrationTools/DatabaseMigrationTools.csproj:68]: TODO: I'll update later with stable version later.

File: src/NuGetGallery.Core/NuGetGallery.Core.csproj [hunk 3]
```diff
@@ -51,7 +51,7 @@
       <Version>6.4.2</Version>
     </PackageReference>
     <PackageReference Include="NuGet.Services.FeatureFlags">
-      <Version>2.109.0</Version>
+      <Version>2.110.0-agr-kv-lib-upgrade3-8361872</Version>
```
Comment (erdembayar) [inline src/NuGetGallery.Core/NuGetGallery.Core.csproj:54]: TODO: I'll update later with stable version later.

File: src/NuGetGallery.Core/NuGetGallery.Core.csproj [hunk 4]
```diff
@@ -60,13 +60,13 @@
   
   <ItemGroup Condition="'$(TargetFramework)' == 'net472'">
     <PackageReference Include="NuGet.Services.Messaging.Email">
-      <Version>2.109.0</Version>
+      <Version>2.110.0-agr-kv-lib-upgrade3-8361872</Version>
```
Comment (erdembayar) [inline src/NuGetGallery.Core/NuGetGallery.Core.csproj:63]: TODO: I'll update later with stable version later.

File: src/GitHubVulnerabilities2Db/GitHubVulnerabilities2Db.csproj [hunk 5]
```diff
@@ -91,7 +91,7 @@
       <Version>4.3.0-dev-8079991</Version>
     </PackageReference>
     <PackageReference Include="NuGet.Services.Cursor">
-      <Version>2.109.0</Version>
+      <Version>2.110.0-agr-kv-lib-upgrade3-8361872</Version>
```
Comment (erdembayar) [inline src/GitHubVulnerabilities2Db/GitHubVulnerabilities2Db.csproj:94]: TODO: I'll update later with stable version later.

File: src/DatabaseMigrationTools/DatabaseMigrationTools.csproj (hunk 1 above)
Comment (erdembayar) [reply src/DatabaseMigrationTools/DatabaseMigrationTools.csproj]: Sorry, about this. I reverted inline comment commit.

**Review by erdembayar — COMMENTED** [review]
Removed inline comments in code

**Review by joelverhagen — APPROVED** [review]

**Pull request merged**